src/object.js

```diff
@@ -371,8 +382,8 @@ TransisObject.prototype.notify = function(event, ...args) {
 //
 // Returns the receiver.
 TransisObject.prototype.didChange = function(name) {
-  (this.__changedProps__ = this.__changedProps__ || {})[name] = true;
-  changedObjects[this.objectId] = this;
+  changedObjects[this.objectId] = changedObjects[this.objectId] || {object: this, props: {}};
```

Comment:
Nit: Maybe consider `registerChange` function as this code is here and in `propagateChanges()`.

```js

function registerChange(object, prop){
  changedObjects[object.objectId] = changedObjects[object.objectId] || {objec, props: {}};
  changedObjects[object.objectId].props[prop] = true;
}

```